### DISCUSSION TITLE
Pass default param value for optional deep link param?

### DISCUSSION THREAD

**Discussion by abelfiore** [react-navigation/react-navigation#10934] [General]
Is there a way, using deep linking, to pass a default value for a parameter that is not provided in the given url? Or can I pass a hard coded value, as a param, that is not part of the link?

**Comment by satya164** [comment]
You can use `initialParams`: https://reactnavigation.org/docs/params/#initial-params

**Reply by abelfiore** [reply]
Can that be done from the linking configuration?

**Reply by satya164** [reply]
The simplest way to do this is using `initialParams` - there shouldn't be a difference between the default params depending on if you're directly navigating or coming from a link. If you want to specify it in the linking config, then you have to manually do it by using a custom `getStateFromPath` function.
